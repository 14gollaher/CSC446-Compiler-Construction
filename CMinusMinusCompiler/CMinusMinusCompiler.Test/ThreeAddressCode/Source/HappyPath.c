int a, b, c, d, f;
const x = 2.2;
int foo()
{
	int a;
	a = x;
	return x;
}

int main()
{
	int z;
	float y;

	const x = 6;

	a = -5;
	b = 10.2;

	b = x;
	c = x;

	d = a;
	f = y;

	z = z;
	c = foo();
	c = foo(a, x, y);
	d = foo(1, 1.25);
	return a;
}

