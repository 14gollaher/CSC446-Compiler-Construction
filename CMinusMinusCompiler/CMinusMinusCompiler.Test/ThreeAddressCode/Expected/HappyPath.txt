PROC foo
	_BP-4 = 2.2
	_BP-2 = _BP-4
	_BP-8 = 2.2
	_AX = _BP-8
ENDP foo

PROC main 
	_BP-8 = -5
	a = _BP-8
	_BP-10 = 10.2
	b = _BP-10
	_BP-14 = 6
	b = _BP-14
	_BP-16 = 6
	c = _BP-16
	d = a
	f = _BP-4
	_BP-2 = -_BP-2
	_CALL foo
	c = _AX
	_PUSH _BP-4
	_BP-18 = 6
	_PUSH _BP-18
	_PUSH a
	_CALL foo
	c = _AX
	_BP-20 = 1.25
	_PUSH _BP-20
	_BP-24 = 1
	_PUSH _BP-24
	_CALL foo
	d = _AX
	_AX = a
ENDP main

START PROC main